deploy/supervisor/values.yaml

```diff
@@ -73,3 +73,43 @@ api_group_suffix: pinniped.dev
 #! Optional.
 https_proxy: #! e.g. http://proxy.example.com
 no_proxy: "$(KUBERNETES_SERVICE_HOST),169.254.169.254,127.0.0.1,localhost,.svc,.cluster.local" #! do not proxy Kubernetes endpoints
+
+#! Control the https and http listeners of the Supervisor.
+#!
+#! The schema of this config is as follows:
+#!
+#! endpoints:
+#!   https:
+#!     network: tcp | unix | disabled
+#!     address: interface:port when network=tcp or /pinniped_socket/socketfile.sock when network=unix
+#!   http:
+#!     network: same as above
+#!     address: same as above
+#!
+#! Setting network to disabled turns off that particular listener.
+#! See https://pkg.go.dev/net#Listen and https://pkg.go.dev/net#Dial for a description of what can be
+#! specified in the address parameter based on the given network parameter.  To aid in the use of unix
+#! domain sockets, a writable empty dir volume is mounted at /pinniped_socket when network is set to "unix."
+#!
+#! The current defaults are:
+#!
+#! endpoints:
+#!   https:
+#!     network: tcp
+#!     address: :8443
+#!   http:
+#!     network: tcp
+#!     address: :8080
+#!
+#! These defaults mean: bind to all interfaces using TCP.  Use port 8443 for https and 8080 for http.
+#! The defaults will change over time.  Users should explicitly set this value if they wish to avoid
+#! any changes on upgrade.  The http listener will be disabled by default in a future release.
```

Comment:
Rather than saying that the http listener will be disabled in a future release, it might be better to use the same language that we used in the docs which doesn't commit to it being disabled: "A future version of the Supervisor app may include a breaking change to adjust the default behavior of the HTTP port to only listen on 127.0.0.1 (or perhaps even to be disabled)".

The reason that we might choose to bind the http port to localhost by default in the future (as opposed to disabled by default) is that it would allow the configuration of service mesh sidecars using http from within the same pod by default. It shouldn't hurt to have the http listener listening within the pod all the time, even if it is unused. It would be effectively unreachable except by sidecars.

Reply:
Updated the language here to match the docs.

> It shouldn't hurt to have the http listener listening within the pod all the time, even if it is unused. It would be effectively unreachable except by sidecars.

If you set host network to `true`, then this listener will automatically get exposed on the node.  Thus to me it is safer to disable the listener and require opt-in to explicitly enable it as needed - then there can never be accidental exposure.  We will get some minor resource savings and service mesh sidecars already require other configurations to get working correctly, so I do not think having to enable this listener explicitly is burdensome.

Reply:
The burden on the user would be that the user would have to switch from using all of our defaults that we ship as yaml manifests attached to our releases on GitHub to instead using ytt to render (or overlay) their own manifests. If the http port were bound to localhost by default then a user could use a service mesh with our default settings.

On the other hand, I suppose it would be safer for host networking situations, that makes sense. Although if you're going to use host networking then our defaults are probably not what you want anymore, so you probably need to start using ytt to render (or overlay) your own manifests at that point anyways, and if you're already using ytt then you can turn off the http port easily thanks to this PR. :)

Reply:
I would prefer to assume that the user does not know that the supervisor has an http listener at all.  It is strange for an OIDC server to have such a listener.

An alternative to disablement would be a default enabled unix domain socket.  That is safe even with host networking because the socket volume is not a host path.

Either way, I am not too concerned about service mesh users having to use ytt to render things.  There are plenty of other knobs that need to be tweaked to get a working production install (with or without a service mesh).  If the service mesh case was truly important to us, we would have automated tests for it.